### PR TITLE
feat: Add header aliasing system for Parse headers

### PR DESCRIPTION
**Pull request by Moumouls**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/parse-server/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/parse-server/blob/alpha/LICENSE).

## Issue

Security hardening through header masking:
According to OWASP recommendations, it is best practice to mask or obfuscate technology-specific details whenever possible. For example, headers such as X-Parse-Session-Token and X-Parse-Application-Id clearly reveal the backend technology stack and should be renamed or aliased to reduce exposure.

Session management in micro frontend architectures or multi parse-server under one domain:
In a micro frontend setup with multiple Parse Server instances under the same domain, developers may not be able to use HttpOnly cookies. As a result, they might be forced to rely on alternative methods, often less secure, such as localStorage or sessionStorage to manage session tokens across instances.
This feature allows developers to configure alias names for session headers and application IDs, helping maintain flexibility while improving security.

## Approach

**Alias system:** redirects the value of a custom (aliased) header to the corresponding internally recognized Parse header.

Made with Composer 2 and Codex 5.3 High + manually reduced implementation complexity

## Tasks

- [x] Add tests
- [x] Add changes to documentation (guides, repository pages, code comments)
- [ ] Add [security check](https://github.com/parse-community/parse-server/blob/master/CONTRIBUTING.md#security-checks)
- [ ] Add new Parse Error codes to Parse JS SDK 

## Summary by CodeRabbit

* **New Features**
  * Added `headerAliases` support for selected Parse headers across REST, GraphQL, CORS, and authentication.
  * Added GraphQL CSRF support for configured header aliases.
* **Bug Fixes**
  * Ensured deterministic alias selection and canonical-header precedence.
  * Included configured aliases in CORS allowed headers.
* **Documentation**
  * Documented configuration, environment-variable usage, supported headers, restrictions, and defaults.
* **Tests**
  * Expanded validation, security, precedence, and edge-case coverage.

### CODE REVIEW

**Comment by Moumouls** [comment]
@coderabbitai do you see security issues ? Or strong implementation gaps ?

**Comment by mtrezza** [comment]
@Moumouls Is this ready for review? I see an open CR feedback...

File: src/middlewares.js [hunk 1]
```diff
@@ -65,6 +65,59 @@ export const checkIp = (ip, ipRangeList, store) => {
...
+          .map(alias => [alias.toLowerCase(), source.toLowerCase()])
+      )
+      .flat()
+  );
+  Object.entries(req.headers).forEach(([header, value]) => {
+    const targetHeader = indexHeaderByAlias[header.toLowerCase()];
+    if (targetHeader) {
+      req.headers[targetHeader] = value;
+    }
+  });
```
[the remark this answers is not in the document]
Comment (mtrezza) [reply src/middlewares.js:110]: Check to see whether the issue is addressed.

**Comment by mtrezza** [comment]
Rebased...

File: src/middlewares.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mtrezza) [reply src/middlewares.js:110]: @Moumouls What's the expected behavior? Seems like an edge case...

**Comment by Moumouls** [comment]
@coderabbitai full review

**Comment by Moumouls** [comment]
@coderabbitai full review

**Comment by Moumouls** [comment]
Tracking this one on my week todo list @mtrezza

**Comment by Moumouls** [comment]
Do you want me to check specific things ? @mtrezza

**Comment by mtrezza** [comment]
@Moumouls there's a bunch of open review feedback

**Comment by Moumouls** [comment]
@coderabbitai full review

**Comment by Moumouls** [comment]
@mtrezza so i removed as suggested the master key alias, it's too dangerous, and i don't see a practical use case where master key should be aliased

**Comment by Moumouls** [comment]
@coderabbitai full review

**Comment by Moumouls** [comment]
@coderabbitai full review

**Comment by Moumouls** [comment]
@coderabbitai full review

**Comment by Moumouls** [comment]
@mtrezza tell me what you think here
